Library/WDIScatter.Rmd

```diff
@@ -0,0 +1,115 @@
+---
+# Use a description of your plot here to facilitate search
+pagetitle: "Scatter Plots: Plot Multiple Development Indicators against GDP Per Capita using WDI"
+
+# Add your GitHub handle to get credit for your contribution
+author: "@yining-w"
+
+# Date is optional
+date: "09/12/2022"
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r SameNameAsYourScript, fig.path = '../docs/figure/', warning = FALSE}
+
+# Install and load packages ===========================================================
+
+packages <- 
+  c("tidyverse",
+    "ggplot2",
+    "ggrepel",
+    "WDI")
+
+pacman::p_load(packages, 
+               character.only = TRUE, 
+               install = FALSE) # Change to install = TRUE to install the required packages
+
+# Data wrangling =====================================================================
+
+# Data is called directly using WDI()
+
+# Search for indicators to call in WDI
+ind <- WDIsearch(string='ENRL', field="indicator", cache=NULL)
+indicators = ind[,1]
+
+# Get indicators 
+df <- WDI(country = "all",
+  indicator = c("NY.GDP.PCAP.KD", indicators),
+  extra = TRUE,
+  cache = NULL,
+  latest = 1, 
+  start = 2012,
+  language = "en")
+
+
+# First, drop non country level aggregates
+df <- df %>% filter(!is.na(region))
+
+## Keep a list of indicators that were successfully called from WDI
+cols <- colnames(df)
+cols <- cols[-(1:5)]
+cols <- cols[-(28:34)]
+
+
+## get the variable labels of indicators successfully called, merge the data labels
+df_ind <- merge(data.frame(indicators), data.frame(cols), by.x = "indicator", by.y = "cols")
```

Comment:
this should be corrected to
```
df_ind <- merge(data.frame(indicators), data.frame(cols), by.x = "indicators", by.y = "cols")
```
note the different name in the variable in `by.x`

Reply:
Hi Luis, I wanted to focus on creating graphs through a loop to 1) query multiple searches in a loop through WDI and 2) graph the results in a loop. I see your point though, happy to submit something else in the future.